canonical/library.canonical.com
# Write tests for the Navigation Class

## Done

Adds some tests for the different aspects of the navigation class to alleviate possible regressions moving forward with the projects. These include:
- testing slug transformation
- testing unwanted items aren't added to the navigation
- testing top level items are attached to the top level of the navigation
- testing children are attached to their parents in the navigation `dict`
- testing top level items are ordered properly

Refactors how we handle the mock drive by creating a custom class that returns mock data, rather than monkey patching the mocked data in

## QA

- Check the code for any oversights in the tests or false passes
- Check all tests are passing


## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5382
